src/NanoMsg.cpp

```diff
@@ -0,0 +1,49 @@
+#include "NanoMsg.h"
+#include <cstring>
+#include <stdexcept>
+#include <g3log/g3log.hpp>
+/**
+ * Constructor for copying a packet hash vector into a nanomsg buffer
+ */
+NanoMsg::NanoMsg(PacketHashVector& data) : sent(false) {
+   auto data_size = data.size() * sizeof(std::pair<void*, unsigned int>);
+   buffer = nn_allocmsg(data_size, 0);
```

Comment:
1) ```0``` is very hard to understand @yannick-polius. 
Please instead put a named variable 
```
const int defaultAllocationPolicy = {0};
buffer = nn_allocmsg(data_size, 0);
```
ref: http://nanomsg.org/v0.2/nn_allocmsg.3.html